README.md

```diff
@@ -1,125 +1,167 @@
-# mason-null-ls
+`mason-null-ls` bridges `mason.nvim` with the `null-ls` plugin - making it easier to use both plugins together.
 
-Install or upgrade the required tools needed for null-ls where available in Mason.
 
-Currently installation of the tools must occur using a command on startup. 
+# Introduction
 
-`require("mason-null-ls").check_install()`
+`mason-null-ls.nvim` closes some gaps that exist between `mason.nvim` and `null-ls`. Its main responsibilities are:
 
-Uses Mason do the heavy lifting of installing the dependencies.
+-   provide extra convenience APIs such as the `:NullLsInstall` command
+-   allow you to (i) automatically install, and (ii) automatically set up a predefined list of sources
+-   translate between `null-ls` source names and `mason.nvim` package names (e.g. `haml_lint` <-> `haml-lint`)
 
-**Can automatically parse the null-ls table and install the necessary tools available in Mason.**
+It is recommended to use this extension if you use `mason.nvim` and `null-ls`.
 
-Note: Going through all null-ls builtins and mapping it to the correct tool name in Mason is still a WIP.
-Feel free to create an issue or PR with a source you would like available immediately.
+**Note: this plugin uses the `null-ls` source names in the APIs it exposes - not `mason.nvim` package names.
 
-## Dependencies
 
-- [mason.nvim](https://github.com/williamboman/mason.nvim.git)
-- [null-ls.nvim](https://github.com/jose-elias-alvarez/null-ls.nvim.git)
-- Neovim >0.7
+# Requirements
 
-## Installation
+-   neovim `>= 0.7.0`
+-   `mason.nvim`
+-   `null-ls.nvim`
 
-### Packer Example
+
+# Installation
+
+## [Packer](https://github.com/wbthomason/packer.nvim)
 
 ```lua
 use {
-	"jayp0521/mason-null-ls.nvim",
-	after = {
-		"null-ls.nvim",
-		"mason.nvim",
-	},
+    "williamboman/mason.nvim",
+    "jayp0521/mason-null-ls.nvim",
+    "jose-elias-alvarez/null-ls.nvim",
 }
 ```
 
-### Recommended
-```lua
-use {
-	"jayp0521/mason-null-ls.nvim",
-	after = {
-		"null-ls.nvim",
-		"mason.nvim",
-	},
-	config = function()
-		require("mason-null-ls").setup({
-			automatic_installation = true,
-		})
-		require("mason-null-ls").check_install(true)
-	end,
-}
+## vim-plug
+
+```vim
+Plug 'williamboman/mason.nvim'
+Plug 'jayp0521/mason-null-ls.nvim'
+Plug 'jose-elias-alvarez/null-ls.nvim'
 ```
```

Comment:
why suggest `vim-plug`. people want lua when using neovim

Reply:
Figured some people would be still using it, feel free to modify the README however you see fit.